optuna/pruners/hyperband.py

```diff
@@ -150,10 +150,15 @@ def prune(self, study: "optuna.study.Study", trial: FrozenTrial) -> bool:
         bracket_study = self._create_bracket_study(study, i)
         return self._pruners[i].prune(bracket_study, trial)
 
-    # TODO(crcrpar): Improve resource computation/allocation algorithm.
-    def _calc_bracket_resource_budget(self, pruner_index: int, n_brackets: int) -> int:
-        n = self._reduction_factor ** (n_brackets - 1)
-        return n + (n / 2) * (n_brackets - 1 - pruner_index)
+    def _calc_trial_allocation_budget(self, pruner_index: int) -> int:
+        """Computes the trial allocated budget for a bracket of ``pruner_index``.
+
+        In the `original paper <http://www.jmlr.org/papers/volume18/16-558/16-558.pdf>`, the
+        number of trials per one bracket is referred as ``n`` as calculated in Algorithm 1.
```

Comment:
How about explaining that the return value computed by this method, i.e. "n", is merely used to define the split ratio of trials between brackets, and not the actual number of trials per bracket?

Reply:
[super nit] Two "as" ("... is referred as ``n`` as ...") are so close that make this sentence less readable to me.